docs/benchmark/allreduce/report.md

```diff
@@ -0,0 +1,101 @@
+# Benchmark: ElasticDL Calling the Fault-Tolerable AllReduce from Elastic Horovod
+
+We usually use synchronous distributed SGD to train deep learning models in CV
+and NLP. fault tolerance and Elastic scheduling can improve the resource utilization
+and user experience. Horovod has released fault-tolerance AllReduce in v0.20.0.
+ElasticDL implements elastic synchronous training on Kubernetes by calling
+Horovod. So, we experiment to verify the performance of ElasticDL.
+
+## Experiment 1: Elastic Scheduling of Multiple ElasticDL Jobs
+
+Supposed the GPU requirement of two jobs is a little over the total of the
+cluster. Without elastic scheduling, the latter job cannot start until
+another job completes. There are two problems:
+
+- Bad user experience. The submitter of the latter job must wait a long time.
+- Low resource utilization. The rest of GPU in the cluster is free.
+
+Using elastic scheduling, the latter job can start using the rest of GPU and
+does not need to wait for the completion of the earlier job.
+After the earlier job completes, the latter job can scale up to speed up
+training. So, elastic scheduling can improve user experience and
+resource utilization.
+
+To verify the benefit of elastic scheduling, we perform an experiment on a
+Kubernetes cluster with GPU. The experiment result is shown in the following
+figure.
+
+![overlap jobs](./data/experiment_1.png)
+
+In the upper figure, we use gang scheduling to submit two training jobs one
+after another, and each job needs 4 GPUs. However, we create a cluster
+only with 6 GPUs using Alibaba Cloud Container Service for Kubernetes (ACK).
+So, we cannot simultaneously run those two jobs on the cluster.
+The latter cannot start until the earlier job completes.
+We can find the earlier job completes at 465s. Then,
+the latter job starts at 518s and ends at 968s. After the
+earlier job completes, the latter job waits for 53s because ElasticDL must
+launch the master pod firstly. The master pod does not need GPU,
+so the GPU utilization is 0 before the master launches worker pods.
+
+In the figure below, we use ElasticDL to submit the two same jobs.
+We submit the latter job at 120s, and the latter job starts immediately.
+All 6 GPUs on the cluster are used. After the earlier job ends at 474s,
+the latter job scale up to 4 GPUs and ends at 809s. Due to elastic scheduling,
+the elapsed time is less than gang scheduling.
+
+## Experiment 2: AI Training Jobs and Serving Jobs are Running on a GPU Cluster
```

Comment:
I prefer Serving service instead of serving jobs.
The job has limited execution time but the service is long running. The latter one is more suitable for serving scenario.